docs/Makefile

```diff
@@ -2,8 +2,8 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -w warnings.txt
-SPHINXBUILD   = python -msphinx
+SPHINXOPTS    = -w warnings.txt -n -W
```

Comment:
Here the `-n` makes sure that links are correct and the `-W` converts all warnings to errors. If the `-W` is not added then the errors that @zhemao found in the X PR would be ignored.